spec/helpers/vita_partner_helper_spec.rb

```diff
@@ -2,23 +2,115 @@
-
-    it "returns array grouped by parent org" do
-      @vita_partners = VitaPartner.all
-      expected =
-        [
-          ["First Parent Org", [["First Parent Org", parent_org1.id], ["The First Child Org", sub_org1.id], ["The Second Child Org", sub_org2.id]]],
-          ["No Child Org", [["No Child Org", parent_org3.id]]],
-          ["Second Parent Org", [["Second Parent Org", parent_org2.id], ["The Third Child Org", sub_org3.id]]],
+    context "when user's role is Team Member" do
+      let(:team_member) { create :team_member_user }
+
+      before do
+        allow(view).to receive(:current_user).and_return(team_member)
+      end
+
+      it "returns just the user's site and the org" do
+        @vita_partners = VitaPartner.accessible_by(Ability.new(team_member))
+        expected = [
+          [
+            team_member.role.site.parent_organization.name,
+            [[team_member.role.site.name, team_member.role.site.id]]
+          ]
+        ]
+
+        expect(helper.grouped_organization_options).to eq(expected)
+      end
+    end
+
+    context "when the user's role is Site Coordinator" do
+      let(:site_coordinator) { create :site_coordinator_user }
+
+      before do
+        allow(view).to receive(:current_user).and_return(site_coordinator)
+      end
+
+      it "returns just the user's site and the org" do
+        @vita_partners = VitaPartner.accessible_by(Ability.new(site_coordinator))
+        expected = [
+          [
+            site_coordinator.role.site.parent_organization.name,
+            [[site_coordinator.role.site.name, site_coordinator.role.site.id]]
+          ]
         ]
-      expect(helper.grouped_organization_options).to eq(expected)
+
+        expect(helper.grouped_organization_options).to eq(expected)
+      end
     end
 
+    context "when the user's role is Admin" do
+      let(:parent_org1) { create(:vita_partner, name: "First Parent Org") }
+      let(:parent_org2) { create(:vita_partner, name: "Second Parent Org") }
+      let(:parent_org3) { create(:vita_partner, name: "No Child Org") }
+      let(:sub_org1) { create(:vita_partner, parent_organization_id: parent_org1.id, name: "The First Child Org") }
+      let(:sub_org2) { create(:vita_partner, parent_organization_id: parent_org1.id, name: "The Second Child Org") }
+      let(:sub_org3) { create(:vita_partner, parent_organization_id: parent_org2.id, name: "The Third Child Org") }
+
+      let(:admin) { create :admin_user }
+
+      before do
+        allow(view).to receive(:current_user).and_return(admin)
+      end
+
+      it "returns arrays of sites grouped by parent orgs" do
+        @vita_partners = VitaPartner.accessible_by(Ability.new(admin))
+        expected =
+          [
+            ["First Parent Org", [["First Parent Org", parent_org1.id], ["The First Child Org", sub_org1.id], ["The Second Child Org", sub_org2.id]]],
```

Comment:
This contains the old (November 2020) style of "Parent Org" and "Sub Organization." Nowadays (Jan 2021) we call them "Organizations" and "Sites." Consider using the new nomenclature. Not a blocker.